### ISSUE TITLE
Low referrals from in product notification

### ISSUE THREAD

**Issue by ckprice** [mozilla/testpilot#1285]
We deployed a push notification for No More 404's on August 22nd.

Between then and now, GA is showing just 2 referrals from Campaign = `push notification`.

This number seems low, and this issue is opened to investigate and fix any issues.

**Comment by lmorchard** [comment]
Mentioned this a few times in IRC & meetings, but will jot it down here for reference:

[There's code in the add-on that ignores the first batch of notifications it sees, if it's never seen notifications before.](https://github.com/mozilla/testpilot/blob/master/addon/lib/experiment-notifications.js#L54) This is mainly so that new installs in the future don't spam users by catching up with old notifications.

However, since we _just_ launched the notification code in the add-on, nearly everyone will have a blank record of notifications and so most likely ignore the first batch. 

So, TL;DR: We might want to try sending out a second notification shortly and see what happens.

**Comment by ghost** [comment]
It sounds like there are some questions around the code that tries to de-dupe and not show the same notification twice.  A separate issue being filed for that.

**Comment by lmorchard** [comment]
Also I think I'm going to try writing some unit tests for the notifications code, just to backfill and try to prove out it's doing what I think it's doing

**Comment by lmorchard** [comment]
Okay, unit tests written in PR #1297 and I didn't find any surprises in the way I expect the code is be working. Also considered Issue #1296 - that's related to changing add-on environment and shouldn't directly affect normal production users.

So, TL;DR redux: I think my TL;DR [from earlier](https://github.com/mozilla/testpilot/issues/1285#issuecomment-242534022) still applies - the easiest "fix" for this is to send out another notification because the first one got ignored by all production users' add-ons.

**Comment by ckprice** [comment]
Okay, so we should send out another No More 404's push notification (I'll add to public calendar).

> However, since we just launched the notification code in the add-on, nearly everyone will have a blank record of notifications and so most likely ignore the first batch. 

For clarification: If a brand new user installs Test Pilot today, and we send a push notification on Monday, will that user see the push notification?

**Comment by lmorchard** [comment]
> For clarification: If a brand new user installs Test Pilot today, and we send a push notification on Monday, will that user see the push notification?

Yes, because a brand new user today will see last week's notification, ignore it, and then see everything past then as new

**Comment by ckprice** [comment]
Created a new notification for 2 PM PST today. As noted in channel, these will start appearing up to 12 hours after this time.

Going to leave this open for a bit until we see traffic coming in the next couple days.

**Comment by ckprice** [comment]
Ref: #1309

**Comment by johngruen** [comment]
Note to self: fuzz stage on monday

**Comment by ghost** [comment]
⛽️
